src/Network/WebSockets/TSSocketManager.m

```diff
@@ -317,11 +383,17 @@ - (void)backgroundConnectTimerExpired {
 }
 
 - (void)backgroundTimeExpired {
-    [[self class] resignActivity];
+    OWSAssert([NSThread isMainThread]);
+    
+    if ([UIApplication sharedApplication].applicationState != UIApplicationStateActive) {
+        [[self class] resignActivity];
```

Comment:
This was a key bug. The app was often closing its websocket after being open for 15 seconds.